x11-drivers/nvidia-drivers/files/bin/setup-nvidia

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+. /usr/share/flatcar/release
+. /usr/share/flatcar/update.conf
+
+NVIDIA_DRIVER_VERSION=450.80.02
+NVIDIA_PRODUCT_TYPE=tesla
+NVIDIA_DOWNLOAD_BASEURL="https://us.download.nvidia.com/${NVIDIA_PRODUCT_TYPE}/"
+NVIDIA_DRIVER_BASENAME="NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}"
+NVIDIA_WORKDIR='nvidia-workdir'
+
+FLATCAR_DEVELOPER_CONTAINER="flatcar_developer_container.bin"
+FLATCAR_DEVELOPER_CONTAINER_URL=""
+FLATCAR_ROOT_WORKDIR="/opt/nvidia/workdir"
+FLATCAR_KERNEL_VERSION=$(uname -r)
+NVIDIA_FLATCAR_VERSION_PAIR="${NVIDIA_DRIVER_VERSION}/${FLATCAR_KERNEL_VERSION}"
+NVIDIA_CURRENT_INSTALLATION="current"
+
+function download_flatcar_developer_container() {
+  echo Downloading Flatcar Container Linux Developer Container for version: "${FLATCAR_RELEASE_VERSION}"
+
+  if [[ ! -f "${FLATCAR_ROOT_WORKDIR}/${FLATCAR_DEVELOPER_CONTAINER}" ]]
+  then
+
+    if [[ "$GROUP" == "developer" ]]
+    then
+      FLATCAR_DEVELOPER_CONTAINER_URL="https://storage.googleapis.com/flatcar-jenkins/developer/developer/boards/${FLATCAR_RELEASE_BOARD}/${FLATCAR_RELEASE_VERSION}/${FLATCAR_DEVELOPER_CONTAINER}.bz2"
+    else
+      FLATCAR_DEVELOPER_CONTAINER_URL="https://${GROUP}.release.flatcar-linux.net/${FLATCAR_RELEASE_BOARD}/${FLATCAR_RELEASE_VERSION}/${FLATCAR_DEVELOPER_CONTAINER}.bz2"
+    fi
+
+    if [ ! -n "${FLATCAR_DEVELOPER_CONTAINER_URL}" ]
+    then
+      return 1
+    fi
+
+    curl -L "${FLATCAR_DEVELOPER_CONTAINER_URL}" -o "${FLATCAR_ROOT_WORKDIR}/${FLATCAR_DEVELOPER_CONTAINER}.bz2"
+    bzip2 -d "${FLATCAR_ROOT_WORKDIR}/${FLATCAR_DEVELOPER_CONTAINER}.bz2"
+
+  fi
+
+  return 0
+}
+
+function download_nvidia_driver_archive() {
+  echo Downloading NVIDIA "${NVIDIA_DRIVER_VERSION}" Driver
+
+  if [ ! -f "${FLATCAR_ROOT_WORKDIR}/${NVIDIA_WORKDIR}/${NVIDIA_DRIVER_BASENAME}.run" ]
```

Comment:
This checks for the same file that is the download destination, not catching uncompleted downloads. I suggest to first download to a temporary location (add a `.tmp` suffix) and move it after the download completed.

Reply:
Sure, will update the logic.

Reply:
I've added the flags, so will integrate with the next update.